### PR TITLE
feat: batch API for historical, aggregated, and latest value data

### PR DESCRIPTION
**Pull request by boweihan**
## Overview
Merging `feature-batch` into mainline.

Switch from using single asset property APIs to batch APIs for fetching time-series data. Widget requests are batched on a per-tick basis by default, with the option of configuring a custom batch duration. 

Associated PRs:
https://github.com/awslabs/iot-app-kit/pull/130 - documentation
https://github.com/awslabs/iot-app-kit/pull/128 - batchDuration setting
https://github.com/awslabs/iot-app-kit/pull/125 - latest value batching
https://github.com/awslabs/iot-app-kit/pull/114 - aggregated data batching
https://github.com/awslabs/iot-app-kit/pull/108 - edge case handling for historical data batching
https://github.com/awslabs/iot-app-kit/pull/102 - historical data batching
https://github.com/awslabs/iot-app-kit/pull/101 - dataloader proxy

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

**Review by diehbria — CHANGES_REQUESTED** [review]

File: docs/AWSIoTSiteWiseSource.md [hunk 1]
```diff
@@ -206,3 +211,28 @@ Type: Boolean
...
+import { initialize } from '@iot-app-kit/source-iotsitewise';
+
+const { IoTSiteWiseClient } = require("@aws-sdk/client-iotsitewise");
+
+const iotsitewiseClient = new IoTSiteWiseClient({ region: "REGION" });
+
+const { query } = initialize({ iotsitewiseClient, settings: { batchDuration: 100 } });
+```
+
+`batchDuration`
```
Comment (diehbria) [inline docs/AWSIoTSiteWiseSource.md:230]: the preceding paragraph should be re-arranged a little. The very first word should be (Optional) [basic description]

**Review by diehbria — CHANGES_REQUESTED** [review]
Comments on documentation for consistency

File: packages/components/src/testing/testing-ground/testing-ground.tsx [hunk 2]
```diff
@@ -56,32 +54,111 @@ export class TestingGround {
...
+                      properties: [{ propertyId: DEMO_TURBINE_ASSET_2_PROPERTY_4 }],
                     },
                   ],
                 }),
               ]}
             />
-            <iot-line-chart
-              widgetId="line-2"
+            <iot-scatter-chart
+              widgetId="kpi-1"
```
Comment (NorbertNader) [inline packages/components/src/testing/testing-ground/testing-ground.tsx:100]: nit: scatter

File: packages/components/src/testing/testing-ground/testing-ground.tsx [hunk 3]
```diff
@@ -56,32 +54,111 @@ export class TestingGround {
           <br />
           <br />
           <div style={{ width: '400px', height: '500px' }}>
-            <iot-scatter-chart
-              widgetId="scatter-1"
-              viewport={{ duration: '5m', yMin: 0, yMax: 1 }}
+            <iot-line-chart
+              widgetId="kpi-1"
```
Comment (NorbertNader) [inline packages/components/src/testing/testing-ground/testing-ground.tsx:58]: nit: line

**Review by NorbertNader — APPROVED** [review]

**Pull request merged**